py-polars/polars/frame.py

```diff
@@ -724,6 +725,52 @@ def dtypes(self) -> "List[type]":
         """
         return [datatypes.dtypes[idx] for idx in self._df.dtypes()]
 
+    def describe(self):
+        """
+        Summary statistics for a DataFrame. Only summarizes numeric datatypes at the moment and returns nulls.
+
+        Example
+        ---
+        ```python
+        >>> dataframe = pl.DataFrame({
+            'a': [1.0, 2.8, 3.0],
+            'b': [4, 5, 6],
+            "c": [True, False, True]
+            })
+        >>> dataframe.describe()
+        shape: (4, 4)
+        ╭──────────┬───────┬─────┬───────╮
+        │ describe ┆ a     ┆ b   ┆ c     │
+        │ ---      ┆ ---   ┆ --- ┆ ---   │
+        │ str      ┆ f64   ┆ f64 ┆ f64   │
+        ╞══════════╪═══════╪═════╪═══════╡
+        │ "mean"   ┆ 2.267 ┆ 5   ┆ 0.667 │
+        ├╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌┤
+        │ "std"    ┆ 1.102 ┆ 1   ┆ 0.577 │
+        ├╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌┤
+        │ "min"    ┆ 1     ┆ 4   ┆ 0.0   │
+        ├╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌┼╌╌╌╌╌╌╌┤
+        │ "max"    ┆ 3     ┆ 6   ┆ 1     │
+        ╰──────────┴───────┴─────┴───────╯
+        """
+        columns = []
+
+        for s in self:
```

Comment:
This is unneeded expensive. You now cast the whole Series (which my be millions of elements), duplicating memory in the process.

This cast is needed to have the same types in the resulting `DataFrame`, so it's better to cast the results, they contain only a single row. 

Maybe add a comment on why this coercion is needed, as this may be forgotten next time we glance at this. :)